lib/util/cache/package/decorator.ts

```diff
@@ -43,124 +47,129 @@ interface CacheParameters {
-      // data as obsolete according to the "soft" TTL while physically storing it
-      // according to the "hard" TTL.
-      //
-      // This helps us return obsolete data in case of upstream server errors,
-      // which is more useful than throwing exceptions ourselves.
-      //
-      // However, since the default hard TTL is one week, it could create
-      // unnecessary pressure on storage volume. Therefore,
-      // we cache only `getReleases` and `getDigest` results for an extended period.
-      //
-      // For other method names being decorated, the "soft" just equals the "hard" ttl.
-      if (methodName !== 'getReleases' && methodName !== 'getDigest') {
-        hardTtlMinutes = softTtlMinutes;
-      }
-
-      let oldData: unknown;
-      if (oldRecord) {
-        const now = DateTime.local();
-        const cachedAt = DateTime.fromISO(oldRecord.cachedAt);
-
-        const softDeadline = cachedAt.plus({ minutes: softTtlMinutes });
-        if (now < softDeadline) {
-          return oldRecord.value;
+        let {
+          // eslint-disable-next-line prefer-const
+          softTtlMinutes,
+          hardTtlMinutes,
+        } = resolveTtlValues(finalNamespace, ttlMinutes);
+
+        // The separation between "soft" and "hard" TTL allows us to treat
+        // data as obsolete according to the "soft" TTL while physically storing it
+        // according to the "hard" TTL.
+        //
+        // This helps us return obsolete data in case of upstream server errors,
+        // which is more useful than throwing exceptions ourselves.
+        //
+        // However, since the default hard TTL is one week, it could create
+        // unnecessary pressure on storage volume. Therefore,
+        // we cache only `getReleases` and `getDigest` results for an extended period.
+        //
+        // For other method names being decorated, the "soft" just equals the "hard" ttl.
+        // We only extend the cache for `getReleases` and `getDigest` because their results
+        // are considered final, whereas other cached methods are considered to work with intermediate values.
+        if (methodName !== 'getReleases' && methodName !== 'getDigest') {
+          hardTtlMinutes = softTtlMinutes;
         }
 
-        const hardDeadline = cachedAt.plus({ minutes: hardTtlMinutes });
-        if (now < hardDeadline) {
-          oldData = oldRecord.value;
+        let fallbackValue: unknown;
+        if (cachedRecord) {
+          const now = DateTime.local();
+          const cachedAt = DateTime.fromISO(cachedRecord.cachedAt);
+
+          if (now < cachedAt.plus({ minutes: softTtlMinutes })) {
+            return cachedRecord.value;
+          }
+
+          if (now < cachedAt.plus({ minutes: hardTtlMinutes })) {
+            fallbackValue = cachedRecord.value;
+          }
         }
-      }
 
-      let newData: unknown;
-      if (oldData) {
+        let newValue: unknown;
```

Comment:
can we extract those renames / refactors to be merged first, so this pr get's smaller?

Reply:
https://github.com/renovatebot/renovate/pull/40014